src/pages/SixCoinPuzzle.js

```diff
@@ -0,0 +1,131 @@
+import "../styles.css";
+import React, { useState } from "react";
+import toast, { Toaster } from "react-hot-toast";
+import ArrowKeysReact from "arrow-keys-react";
+import Button from "../components/Button";
+import Coin from "../components/Coin";
+
+const SixCoinPuzzle = () => {
+  const numberOfCoinSlots = 7;
+  const startingArray = [1, 1, 1, 0, 2, 2, 2];
+  const winningArray = [2, 2, 2, 0, 1, 1, 1];
+  const [coinTypeArray, setCoinTypeArray] = useState(startingArray);
+  const [selectedCoin, setSelectedCoin] = useState(0);
+  const [moveCounter, setMoveCounter] = useState(0);
+  const [coinSelectText, setCoinSelectText] = useState(
+    "Click to select a coin"
+  );
+
+  const swapCoins = (array, index1, index2) => {
```

Comment:
Fun exercise - think you can find a way to make these swapCoins functions (or any others) generic enough that you could use one swapCoins function and pass in different parameters to satisfy both five and six coin puzzles?